### ISSUE TITLE
Couchsurfing app Hangout function doesn't work

### ISSUE THREAD

**Issue by Chrisbuntu** [microg/android_packages_apps_GmsCore#349]
The couchsurfing app has a funcion that uses your current location and it doesn't work anymore in the new versions of service core. Last version working is 0.2.4-64-gcbfb373 with couchsurfing app version 4.9.1-481d0a6. 
It doesn't work with newer version of the microg service core and that version of the couchsurfing app, nor with any newer versions of the couchsurfing or service core app.

**Comment by Chrisbuntu** [comment]
This issue still has not been resolved and now it's not possible to download version 0.2.4-64-gcbfb373 in F-Droid anymore...
Is there any chance this issue can be resolved? Or is there at least any possibility to download version 0.2.4-64-gcbfb373 ? It's the only newer version working with the couchsurfing hangout function.

**Comment by JERW86** [comment]
@Chrisbuntu a logcat of what happens when you try to use the app on the current version (0.2.4-103) and a link to where the app can be downloaded would be helpful to solve this. Here is the version you requested: 
[microGServicesCore-0.2.4-64-gcbfb373.zip](https://github.com/microg/android_packages_apps_GmsCore/files/1083595/microGServicesCore-0.2.4-64-gcbfb373.zip) You can also download older versions of microG by adding https://microg.org/fdroid/archive repo to in the fdroid app.

**Comment by Chrisbuntu** [comment]
@JERW86 I pasted a logcat here: https://pastebin.com/8eW5dwsd

You can download the app on Google Play only (or Yalp Store, I am using that one). 
Any suggestion what could cause the problem?

**Comment by Chrisbuntu** [comment]
Any idea what could be the problem?

**Comment by JERW86** [comment]
@Chrisbuntu No, I'm sorry I don't have the knowledge to help you. But hopefully @mar-v-in or someone else can be of more help.

**Comment by Spetzatin31** [comment]
Still same problem

**Comment by ahstro** [comment]
Is there anything us users can do to aid in fixing this issue? Is the logcat above exhaustive, or do you want or need something different?

Still having this issue with version `0.2.4-111-gf1cdb48`

**Comment by codethief** [comment]
I've been experiencing the same issue, though it did work until a few months ago. (Not sure what the exact versions were that I had installed.)

In any case, this morning I updated the Couchsurfing app through Aurora Store to the most recent version (v4.27.1.50244) and now *I can't open the app at all, anymore*. Every time I try to open it, it tells me "Please install from Google Play" as if it was checking whether I have Google Play installed or not (which I obviously do not). 

This has obviously little to do with MicroG itself and mostly with the fact that the CS app seems to be doing some additional checks these days but I thought I'd mention it, anyway, as a warning.

**Comment by MarS10** [comment]
@codethief its not problem with microg. When you install any app, android (package manager) save information about source (Aurora Store). And CS app just check this information.

**Comment by codethief** [comment]
@MarS10 Oh, I didn't know that the package manager provided this kind of information. That's unfortunate.

> its not problem with microg.

I am aware of that. In fact, I already wrote:

> This has obviously little to do with MicroG itself and mostly with the fact that the CS app seems to be doing some additional checks these days but I thought I'd mention it, anyway, as a warning.

Upon second thought, though, this kind of check that the CS app is doing spoils MicroG's goal of providing an alternative to Googleware in the sense that apps like CS will no longer usable when replacing the Google Services Framework with MicroG. Therefore, maybe this issue actually does fall in the scope of MicroG…? Specifically, how would one go about spoofing an app's source information that Android's package manager is providing to the app itself? (So that apps like CS think they got installed through the Play Store even when in reality they were installed through, say, Aurora.) Would this be achievable at the level of, say, the MicroG-infused LineageOS distribution by adding another patch, similarly to the signature-spoofing one?

**Comment by P3lUZa** [comment]
@codethief you need to modify the xml of the apk, the strategy of CS is to run:
getPackageManager()          .getInstallerPackageName(getPackageName())

And if they find something like com.google.android.packageinstaller
They use the referer to install from gooogle play and abort the program.

Hence the strategy should be to change that package name for com.android.vending

Also, a Russian forum suggests to modify in the XML manifest the main activity to MainActivity instead of WelcomeActivity, I suppose WelcomeActivity is the function running the get installer package name function.

**Comment by P3lUZa** [comment]
Further, I comfirm old apps of CS work with 0.2.4-64-gcbfb373, with lastime versión those apps cant get location on hangouts

**Comment by ale5000-git** [comment]
@Chrisbuntu: Please retry with microG GmsCore 0.2.7

**Comment by P3lUZa** [comment]
0.2.8-17785 still not working
unable to connect to service: com.google.android.gms.appinvite.service.START on com.google.android.gms

**Comment by couraudt** [comment]
It's now fully working with microg version 0.2.9.19420 and couchsurfing app version 4.26.0
Thank you guys!

**Comment by codethief** [comment]
Indeed, I just updated to MicroG v0.2.9.19420 and it's finally working again (using v4.31.0.50259 of the Couchsurfing app).

**Comment by ale5000-git** [comment]
The version 0.2.9 is bugged, please update to 0.2.10

**Comment by codethief** [comment]
> @codethief you need to modify the xml of the apk, the strategy of CS is to run:
> getPackageManager() .getInstallerPackageName(getPackageName())
> 
> And if they find something like com.google.android.packageinstaller
> They use the referer to install from gooogle play and abort the program.
> 
> Hence the strategy should be to change that package name for com.android.vending
> 
> Also, a Russian forum suggests to modify in the XML manifest the main activity to MainActivity instead of WelcomeActivity, I suppose WelcomeActivity is the function running the get installer package name function.

For anyone else facing the same problem: There's the "apps_Packages Info" app on F-Droid that allows starting a particular activity of an app directly.

**Comment by cryptono** [comment]
@codethief Works like charm, thanks for the hint.

To make the changes permanent, you can modify the XML of the app.
For that, you need to change this line:
`<activity android:theme="@android:01030055" android:name="com.couchsurfing.mobile.ui.WelcomeActivity">`
into 
`<activity android:theme="@android:01030055" android:name="com.couchsurfing.mobile.ui.MainActivity">`
It's not hard to modify an APK, see here e.g.: https://www.reddit.com/r/Android/comments/11852r/how_to_modify_an_apk/

**Comment by bo-tato** [comment]
I ran into this problem with latest microg and couchsurfing from aurora store, @codethief solution still works great!
